src/js/tracks/text-track.js

```diff
@@ -336,7 +336,17 @@ class TextTrack extends Track {
    * @param {TextTrack~Cue} cue
    *        The cue to add to our internal list
    */
-  addCue(cue) {
+  addCue(_cue) {
+    let cue = _cue;
+
+    if (!(_cue instanceof window.vttjs.VTTCue)) {
+      cue = new window.vttjs.VTTCue(_cue.startTime, _cue.endTime, _cue.text);
```

Comment:
it's more or less exposed globally right now plus we've expected it to be exposed. For the 5.x version, we definitely should keep it exposed on window because it'll be easier to make it backwards compat.
For this PR, we could look into making a version of vttjs that doesn't bind itself to `window`. Though, I think it should be done as a separate PR rather than part of this work.